segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/ServerConnectionInboundHandler.java

```diff
@@ -84,6 +87,12 @@ public void close() {
             // wait for all messages to be sent before closing the channel.
             ch.eventLoop().execute(() -> ch.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE));
         }
+        isClosed.compareAndSet(false, true);
```

Comment:
can we ensure close() is invoked only once? i.e 
```
if (isClosed.compareAndSet(false, true) ) {
   Channel ch = ...
}
```

Reply:
Ok, I've moved the check as suggested.